### ISSUE TITLE
instrument the remainder of beers-law-lab with tandem

### ISSUE THREAD

**Issue by jbphet** [phetsims/beers-law-lab#99]
The concentration simulation is being instrumented with tandem/together, see https://github.com/phetsims/together/issues/32.  Since concentration is a subset of Beer's Law Lab, we should instrument it as well or decide to leave the beers-law-lab code base in a partially instrumented state.

**Comment by samreid** [comment]
I would recommend to (a) ensure there are no bugs in BLL due to concentration instrumentation, then (b) estimate the amount of work required to instrument then (c) discuss with @kathy-phet for prioritization.

**Comment by pixelzoom** [comment]
Discussed at status meeting. I'll do this to get some first-hand experience with instrumenting a sim.

**Comment by samreid** [comment]
Before @pixelzoom starts on this, it would be good for @jbphet & @samreid to provide some documentation/instruction/support that @pixelzoom or other instrumentors would be able to use.

**Comment by pixelzoom** [comment]
Just to be clear... I'm waiting to start this task until @jbphet provides documentation in phetsims/together#14.

**Comment by pixelzoom** [comment]
Assigning to @jbphet.  Assign back to me when documentation is ready.

**Comment by pixelzoom** [comment]
I created https://github.com/phetsims/tandem/issues/2 for documentation, and assigned to @jbphet.  That issues is a prerequisite to this issue. Assigning this issue back to me.

**Comment by pixelzoom** [comment]
I'm going to start on this without documentation, pairing with @jbphet.

**Comment by pixelzoom** [comment]
@ariel-phet This task is on indefinite hold.  As outlined in https://github.com/phetsims/together/issues/130, together/tandem has a very big hole in its design and implementation.

**Comment by ariel-phet** [comment]
@pixelzoom - understood, I read over the together issue and it seems like a good point. We can discuss with @kathy-phet when she returns. It may be that this refactoring/additional work has to wait until @samreid returns.

**Comment by samreid** [comment]
I factored out the concentration screen together API, please see https://github.com/phetsims/together/issues/130#issuecomment-108606922 for discussion.

**Comment by pixelzoom** [comment]
Unassigning until together/tandem is at a point where this can proceed.

**Comment by samreid** [comment]
At yesterday's phet-io meeting, @kathy-phet suggested getting this ready for work around the beginning of the new year.

We decided on the following plan:
1. I would provide documentation which @pixelzoom can review before starting instrumentation of Beer's Law Lab, and Chris would also review the documentation provided to phet-io clients.
2. @pixelzoom and I would schedule an in-person meeting for @pixelzoom starting on instrumentation

We also discussed preventing this from being an unknown or unlimited time sink by capping it.  @kathy-phet suggested perhaps one week @pixelzoom could spend 8 hours on this and 8 hours on Function Builder.

**Comment by pixelzoom** [comment]
Instrumentation was apparently done by @samreid in https://github.com/phetsims/together/issues/250.

In email to @ariel-phet and @kathy-phet, I asked:
>  I wasn’t aware that the plan was to have Sam do the work, then me re-do the work (presumably without looking at what he’s done). Is there value in having me repeat work that Sam has already done?

@ariel-phet replied:
> You are correct, it would not really make sense for you to redo the work.

... and asked me to (instead) instrument hookes-law, see https://github.com/phetsims/hookes-law/issues/38.

Assigning to @samreid to decide how to proceed with resolving this issue.

**Comment by samreid** [comment]
To proceed, I would like to move the `together-preview` branch to master.  This will put it on equal footing with Concentration and simplify development as we prepare a testable version for Ido.  If @pixelzoom wants to review the instrumentation changes, he will still be able to do so after Hooke's Law has been instrumented.  Any objections?

**Comment by pixelzoom** [comment]
+1 to merge into master.

**Comment by samreid** [comment]
I merged the commits to master.  Would be nice for @pixelzoom to review (a) but not until after working on Hooke's Law instrumentation in Jan 2016 and (b) if time permits.  I'll leave the issue open in case @pixelzoom wants to use this issue to track reviewing.  If @pixelzoom does not want to or does not have time to review, or wants to track the review in another issue, then this issue can be closed.

**Comment by pixelzoom** [comment]
> Would be nice for @pixelzoom to review (a) but not until after working on Hooke's Law instrumentation in Jan 2016 and (b) if time permits.

Sorry, but since it's in master, and it's a sim that I'm responsible for maintaining, I'm going to review the beers-law-lab work now.

**Comment by pixelzoom** [comment]
Related issues noted immediately above.

**Comment by pixelzoom** [comment]
I've reviewed the instrumentation and created issues labeled "phet-io" for any remaining work.  Closing this issue.
